### DISCUSSION TITLE
AsyncWebserver for Portenta H7

### DISCUSSION THREAD

**Discussion by rusty-bit** [khoih-prog/AsyncWebServer_STM32#6] [General]
Hi @khoih-prog,

do you plan to release a AsyncWebserver for the Portenta H7 boards?

**Comment by khoih-prog** [comment]
Hi @rusty-bit 

I currently have no plan yet for Portenta_H7 AsyncWebServer, as the board is still new, and AFAIK there are no underlying library to support Async feature.

I have to research and even write some underlying libraries, such as Portenta_H7_AsyncTCP, Portenta_H7_AsyncUDP, etc. before being able to implement the final Portenta_H7_AsyncWebServer.

It's strange that there still is no STM32H7 Ethernet library ready for Arduino yet.

**Reply by rusty-bit** [reply]
Yes it is strange. Kudo's to you @khoih-prog, to undertake such a task!

**Comment by khoih-prog** [comment]
Already successfully rewrote and modified to support Portenta_H7

Will published new libraries 

1. **Portenta_H7_AsyncTCP**
2. **Portenta_H7_AsyncWebServer**

within several days.



![Selection_048](https://user-images.githubusercontent.com/57012152/136323814-60828100-0e18-4b65-8d0b-f0c124801405.png)

Cheers

**Reply by rusty-bit** [reply]
This is looking great!

**Comment by khoih-prog** [comment]
Check the first library in the series of  Portenta_H7 Async Libraries

1. [**Portenta_H7_AsyncTCP**](https://github.com/khoih-prog/Portenta_H7_AsyncTCP)

Your contribution, by requesting the libraries, has been noted in [**Contributions and Thanks**](https://github.com/khoih-prog/Portenta_H7_AsyncTCP#contributions-and-thanks)

**Comment by khoih-prog** [comment]
Check the second library in the series of  Portenta_H7 Async Libraries

1. [**Portenta_H7_AsyncWebServer**](https://github.com/khoih-prog/Portenta_H7_AsyncWebServer)

Your contribution, by requesting the libraries, again has been noted in [**Contributions and Thanks**](https://github.com/khoih-prog/Portenta_H7_AsyncWebServer#contributions-and-thanks)

**Comment by khoih-prog** [comment]
Check the 3rd library in the series of  Portenta_H7 Async Libraries

1. [**Portenta_H7_AsyncUDP**](https://github.com/khoih-prog/Portenta_H7_AsyncUDP)

**Comment by khoih-prog** [comment]
Check the 4th library in the series of  Portenta_H7 Async Libraries

1. [**Portenta_H7_AsyncHTTPRequest**](https://github.com/khoih-prog/Portenta_H7_AsyncHTTPRequest)
